server/app/repository/ApplicationRepository.java

```diff
@@ -92,9 +92,11 @@ private Application submitApplicationInternal(
       ImmutableList<Application> drafts =
           oldApplications.stream()
               .filter(app -> app.getLifecycleStage().equals(LifecycleStage.DRAFT))
+              .sorted((app1, app2) -> app1.getCreateTime().isAfter(app2.getCreateTime()) ? -1 : 1)
```

Comment:
suggestion (blocking): please add a comment describing the intention of the comparator

Reply:
done.

Reply:
For easier readability, I'd use:

```
.sorted(Comparator.comparing(Application::getCreateTime))
``` 

And if we want the latest first, a call to `reversed()` can be added to the comparator.